STYLEGUIDE.md

```diff
@@ -637,6 +639,66 @@ return (
 - Use `errorMessage` prop to set the error message that is displayed next to the input.
 - If `state='error'` but `errorMessage` is not defined, require `aria-describedby`
 
+## getLgIds
+
+The `getLgIds` utility function generates test IDs for components in the LeafyGreen UI library.
+
+The function accepts an optional root identifier, allowing consumers to set a unique base ID for each instance of a component. This base is then used to generate, scoped test IDs, ensuring predictable targeting in tests.
+
+#### Usage
+
+Each component should export a `getLgIds.ts` file. For more information on naming test IDs, [check out the section on BEM-ish patterns](#follow-bem-ish-patterns-when-hard-coding-a-data-testid-or-data-lgid)
+
+```js
+export const DEFAULT_LGID_ROOT = 'lg-component';
+
+export const getLgIds = (root: `lg-${string}` = DEFAULT_LGID_ROOT) => {
+  const ids = {
+    root,
+    button: `${root}-button`,
+  } as const;
+  return ids;
+};
```

Comment:
maybe mention that the `button` key is defined here because we want to create an id for each "trackable" element in the component. As-written it's not entirely clear why we're using a function vs constants